### PR TITLE
FIXING CID 1324647

### PR DESCRIPTION
**Pull request by AchimTuran**
This PR fixes CID 1324647 and improves my clumsy error message.

### CODE REVIEW

File: lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp [hunk 1]
```diff
@@ -64,10 +64,16 @@ DLLEXPORT void AudioEngine_unregister_me(void *hdl, void* cb)
...
+  {
+    fprintf(stderr, "%s-ERROR: AudioEngine_register_me is called with NULL handle !!!\n", LIBRARY_NAME);
+    return NULL;
+  }
+
   AEStreamHandle *streamHandle = ((CB_AudioEngineLib*)cb)->MakeStream(DataFormat, SampleRate, EncodedSampleRate, Channels, Options);
-  if (!hdl || !cb || !streamHandle)
+  if (!streamHandle)
   {
-    fprintf(stderr, "%s-ERROR: AudioEngine_make_stream can't MakeStrean failed!\n", LIBRARY_NAME);
```
Comment (Jalle19) [inline lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp]: "MakeStrean" should be "MakeStream"

**Comment by Jalle19** [comment]
Should we really use `fprintf` here instead of logging the usual way?

**Comment by AchimTuran** [comment]
> Should we really use fprintf here instead of logging the usual way?

All add-on callback libraries use under https://github.com/xbmc/xbmc/tree/master/lib/addons use fprintf and print their error to stderr. I thinks it's because the functions are embedded to dynamic libraries and easier portability between all platforms.

File: lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp (hunk 1 above)
Comment (AchimTuran) [inline lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp]: > "MakeStrean" should be "MakeStream"

Thanks you're right. Is fixed with the latest commits.

**Comment by MartijnKaijser** [comment]
Doing double work? https://github.com/xbmc/xbmc/pull/8113

**Comment by AchimTuran** [comment]
> Doing double work? #8113

Sorry I didn't recognize this PR.

@ace20022 
If you want, you can take my two commits and integrate them in #8113. It's only an idea.

**Comment by AchimTuran** [comment]
@FernetMenta 
> @AchimTuran coverity is right here, CID 1324647. you first use cb and then you check it for being null

Is it ok from your side?

**Comment by ace20022** [comment]
@AchimTuran it's not worth the hassle ;) I should have pinged you.

**Comment by AchimTuran** [comment]
@ace20022 
Ok. Yeah feel free to ping me on audio signal processing stuff ;-)

**Comment by FernetMenta** [comment]
> Is it ok from your side?

aks coverity :)

**Comment by AchimTuran** [comment]
jenkins build this please

**Pull request merged**